### PR TITLE
AMBARI-25782: Fix syntax, refactor and versions after forwarding from Ambari to ambari-metrics

### PR DESCRIPTION
**Pull request by lucasbak**
## What changes were proposed in this pull request?

Fix syntax, refactor and versions and make `mvn package`  build pass

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -37,14 +37,14 @@
...
+    <slf4j.version>2.0.0</slf4j.version>
     <hbase.tar>https://archive.apache.org/dist/hbase/2.4.2/hbase-2.4.2-bin.tar.gz</hbase.tar>
     <hbase.folder>hbase-2.4.2</hbase.folder>
     <hadoop.tar>https://archive.apache.org/dist/hadoop/common/hadoop-3.1.1/hadoop-3.1.1.tar.gz</hadoop.tar>
     <hadoop.folder>hadoop-3.1.1</hadoop.folder>
     <hadoop.version>3.1.1</hadoop.version>
-    <grafana.folder>grafana-5.2.4</grafana.folder>
-    <grafana.tar>https://s3-us-west-2.amazonaws.com/grafana-releases/release/grafana-5.2.4.linux-amd64.tar.gz</grafana.tar>
+    <grafana.folder>grafana-6.7.4</grafana.folder>
+    <grafana.tar>https://dl.grafana.com/oss/release/grafana-6.7.4.linux-amd64.tar.gz</grafana.tar>
```
Comment (virajjasani) [inline pom.xml:47]: I am not aware of how we can fetch grafana tarball, other than that, changes look good

**Comment by kevinw66** [comment]
There are conflicts after merge #76

**Comment by timyuer** [comment]
`README.md` and `.github/pull_request_template.md` didn't  contain Lincese, This will not pass the RAT check.
@lucasbak

**Comment by lucasbak** [comment]
thanks for the review @timyuer
in which format should I add the license ?
the [License](https://opensource.org/licenses/Apache-2.0) badge is in the README.md

**Comment by timyuer** [comment]
> thanks for the review @timyuer in which format should I add the license ? the [License](https://opensource.org/licenses/Apache-2.0) badge is in the README.md

@lucasbak 

Like this


 in https://github.com/apache/ambari/blob/trunk/README.md?plain=1.

**Comment by lucasbak** [comment]
@timyuer  thanks for the example, I have added in both files the Apache License

**Comment by timyuer** [comment]
+1, test successful in centos7 following `mvn clean install -DskipTests`.

**Comment by timyuer** [comment]
@lucasbak 
Fail to execute `mvn clean install rpm:rpm -DskipTests` in Centos7.

```bash
[INFO] --- rpm-maven-plugin:2.1.4:rpm (default-cli) @ ambari-metrics ---
[INFO] Creating directory /opt/package/code/ambari-metrics/target/rpm/ambari-metrics
[INFO] Creating directory /opt/package/code/ambari-metrics/target/rpm/ambari-metrics/BUILD
[INFO] Creating directory /opt/package/code/ambari-metrics/target/rpm/ambari-metrics/RPMS
[INFO] Creating directory /opt/package/code/ambari-metrics/target/rpm/ambari-metrics/SOURCES
[INFO] Creating directory /opt/package/code/ambari-metrics/target/rpm/ambari-metrics/SPECS
[INFO] Creating directory /opt/package/code/ambari-metrics/target/rpm/ambari-metrics/SRPMS
[INFO] Creating directory /opt/package/code/ambari-metrics/target/rpm/ambari-metrics/tmp-buildroot
[INFO] Creating directory /opt/package/code/ambari-metrics/target/rpm/ambari-metrics/buildroot
[INFO] Creating spec file /opt/package/code/ambari-metrics/target/rpm/ambari-metrics/SPECS/ambari-metrics.spec
[INFO] error: line 4: Empty tag: Release:
[INFO] Building target platforms: noarch-redhat-linux
[INFO] Building for target noarch-redhat-linux
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for ambari-metrics 2.0.0.0-SNAPSHOT:
[INFO] 
[INFO] ambari-metrics ..................................... FAILURE [  2.522 s]
[INFO] Ambari Metrics Common .............................. SKIPPED
[INFO] Ambari Metrics Hadoop Sink ......................... SKIPPED
[INFO] Ambari Metrics Flume Sink .......................... SKIPPED
[INFO] Ambari Metrics Kafka Sink .......................... SKIPPED
[INFO] Ambari Metrics Storm Sink .......................... SKIPPED
[INFO] Ambari Metrics Collector ........................... SKIPPED
[INFO] Ambari Metrics Monitor ............................. SKIPPED
[INFO] Ambari Metrics Grafana ............................. SKIPPED
[INFO] Ambari Metrics Host Aggregator ..................... SKIPPED
[INFO] Ambari Metrics Assembly ............................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.965 s
[INFO] Finished at: 2022-11-29T13:12:56+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:rpm-maven-plugin:2.1.4:rpm (default-cli) on project ambari-metrics: RPM build execution returned: '1' executing '/bin/sh -c cd '/opt/package/code/ambari-metrics/target/rpm/ambari-metrics/SPECS' && 'rpmbuild' '-bb' '--target' 'noarch-redhat-linux' '--buildroot' '/opt/package/code/ambari-metrics/target/rpm/ambari-metrics/buildroot' '--define' '_topdir /opt/package/code/ambari-metrics/target/rpm/ambari-metrics' '--define' '_build_name_fmt %%{ARCH}/%%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm' '--define' '_builddir %{_topdir}/BUILD' '--define' '_rpmdir %{_topdir}/RPMS' '--define' '_sourcedir %{_topdir}/SOURCES' '--define' '_specdir %{_topdir}/SPECS' '--define' '_srcrpmdir %{_topdir}/SRPMS' 'ambari-metrics.spec'' -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Comment by lucasbak** [comment]
@timyuer it's because you need to specify which release it is.
For example for releasing 2.7.0.0
It's from rpm-maven-plugin
```bash
mvn clean install rpm:rpm -Dpackage-release=2.7.7 -DskipTests`
```

**Comment by timyuer** [comment]
> @timyuer it's because you need to specify which release it is. For example for releasing 2.7.0.0 It's from rpm-maven-plugin
> 
> ```shell
> mvn clean install rpm:rpm -Dpackage-release=2.7.7 -DskipTests`
> ```

@lucasbak 

This is a temporary solution, but its default `package-release`  in `pom.xml` seems to have a problem, which needs to be fixed.
And the version for ambari-metrics is `2.0.0.0-SNAPSHOT`.

**Comment by lucasbak** [comment]
It comes from the parse-package-release id plugin
```
regex: ^([0-9]+)\.([0-9]+)\.([0-9]+)\.([0-9]+)(\.|-)((([a-zA-Z]+)?([0-9]+))|(SNAPSHOT)).*
version: 2.0.0.0-SNAPSHOT
replacement: $7 
```
@timyuer I propose we replace the regex by `^([0-9]+)\.([0-9]+)\.([0-9]+)\.([0-9]+)((\.|-)(([0-9]+)|(SNAPSHOT)|(techwin)).*)?` as in `Ambari`.
so replacement $7 matches the release (Snapshot here)

**Comment by timyuer** [comment]
> 

The compilation still fails after the above operations are performed.
Then I found that the version of `rpm-maven-plugin` is not consistent. Is it caused by this problem?
```
[INFO] Creating empty directory /opt/package/code/ambari-metrics/ambari-metrics-timelineservice/target/rpm/ambari-metrics-collector/buildroot/var/lib/ambari-metrics-collector
[INFO] Creating spec file /opt/package/code/ambari-metrics/ambari-metrics-timelineservice/target/rpm/ambari-metrics-collector/SPECS/ambari-metrics-collector.spec
[WARNING] error: License field must be present in package: (main package)
[INFO] Building target platforms: noarch-redhat-linux
[INFO] Building for target noarch-redhat-linux
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for ambari-metrics 2.0.0.0-SNAPSHOT:
[INFO] 
[INFO] ambari-metrics ..................................... SUCCESS [  8.346 s]
[INFO] Ambari Metrics Common .............................. SUCCESS [ 15.320 s]
[INFO] Ambari Metrics Hadoop Sink ......................... SUCCESS [  8.041 s]
[INFO] Ambari Metrics Flume Sink .......................... SUCCESS [  3.666 s]
[INFO] Ambari Metrics Kafka Sink .......................... SUCCESS [  4.242 s]
[INFO] Ambari Metrics Storm Sink .......................... SUCCESS [  3.506 s]
[INFO] Ambari Metrics Collector ........................... FAILURE [ 54.423 s]
[INFO] Ambari Metrics Monitor ............................. SKIPPED
[INFO] Ambari Metrics Grafana ............................. SKIPPED
[INFO] Ambari Metrics Host Aggregator ..................... SKIPPED
[INFO] Ambari Metrics Assembly ............................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:38 min
[INFO] Finished at: 2022-11-29T19:54:28+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:rpm-maven-plugin:2.0.1:rpm (default-cli) on project ambari-metrics-timelineservice: RPM build execution returned: '1' executing '/bin/sh -c cd /opt/package/code/ambari-metrics/ambari-metrics-timelineservice/target/rpm/ambari-metrics-collector/SPECS && rpmbuild -bb --buildroot /opt/package/code/ambari-metrics/ambari-metrics-timelineservice/target/rpm/ambari-metrics-collector/buildroot --define '_topdir /opt/package/code/ambari-metrics/ambari-metrics-timelineservice/target/rpm/ambari-metrics-collector' --target noarch-redhat-linux ambari-metrics-collector.spec' -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :ambari-metrics-timelineservice
```

**Comment by lucasbak** [comment]
For the version, the 2.1.4 in `pom.xml` comes from commit forwarding, we can set `2.1.4` everywhere

**Comment by kevinw66** [comment]
Thanks @lucasbak , compilation succeed now.

**Pull request merged**